src/backend/starknet.cairo

```diff
@@ -110,15 +110,15 @@ namespace Starknet {
         alloc_locals;
         let (block_number) = get_block_number();
         let (block_timestamp) = get_block_timestamp();
-        let (tx_info) = get_tx_info();
         let (coinbase) = Kakarot_coinbase.read();
         let (base_fee) = Kakarot_base_fee.read();
         let (block_gas_limit) = Kakarot_block_gas_limit.read();
         let (prev_randao) = Kakarot_prev_randao.read();
 
         // No idea why this is required - but trying to pass prev_randao directly causes bugs.
         let prev_randao = Uint256(low=prev_randao.low, high=prev_randao.high);
-        let (_, chain_id) = unsigned_div_rem(tx_info.chain_id, 2 ** 32);
+        let (contract_address) = get_contract_address();
+        let (chain_id) = IKakarot.eth_chain_id(contract_address=contract_address);
```

Comment:
we are already in the kakarot contract, so this is calling oneself with a syscall (does it even work)?